pkg/storaged/details.js

```diff
@@ -48,7 +48,24 @@
                 return client.mdraids[path].Start({ "start-degraded": { t: 'b', v: true } });
             },
             mdraid_stop: function mdraid_stop(path) {
-                return client.mdraids[path].Stop({});
+                var mdraid = client.mdraids[path];
+                var block = client.mdraids_block[path];
+                var usage = block? utils.get_usage_alerts(client, block.path) : [ ];
+
+                if (usage.length === 0)
+                    return mdraid.Stop({});
+
+                dialog.open({ Title: cockpit.format(_("Please confirm stopping of $0"),
+                                                    utils.mdraid_name(mdraid)),
```

Comment:
To do this, we would need to know why the stopping has failed.

I agree that it is weird to have a button that sometimes puts up a dialog and sometimes not.  @andreasn, what do you think?